add_tasks.js

```diff
@@ -0,0 +1,198 @@
+    tasks[0].taskName;
+  countTasks();
+}
+
+function deleteCompletedTask() {
+  var containerTaskActive = document.querySelectorAll(".task_active");
+
+  for (var i = 0; i < containerTaskActive.length; i++) {
+    var childTask = containerTaskActive[i].firstElementChild;
+
+    if (childTask.matches(":checked")) {
+      var deleteLabel = containerTaskActive[i].getElementsByClassName(
+        "task_label_checkbox"
+      )[0].innerHTML;
+      var indexDelete = tasks.indexOf(deleteLabel);
+      tasks.splice(indexDelete, 1);
+      var deleteElement = containerTaskActive[i].remove();
+    }
+    countTasks();
+  }
+}
+
+function deleteTask(e) {
+  var containerDeleteTask = e.target.parentNode;
+  var deleteLabel = containerDeleteTask.getElementsByClassName(
+    "task_label_checkbox"
+  )[0].innerHTML;
+  var indexDelete = tasks.indexOf(deleteLabel);
+  tasks.splice(indexDelete, 1);
+  var deleteElement = containerDeleteTask.remove();
+
+  countTasks();
+}
+
+function countTasks() {
+  document.getElementsByClassName("task_label_count")[0].innerHTML =
+    tasks.length;
+}
+
+function showTaskCompleted() {
+
+  showTaskAll();
+  var containerTaskActive = document.querySelectorAll(".task_active");
+
+  for (var i = 0; i < containerTaskActive.length; i++) {
+    var childTask = containerTaskActive[i].firstElementChild;
+
+    if (!childTask.matches(":checked")) {
+      var filterLabel = containerTaskActive[i].getElementsByClassName(
+        "task_label_checkbox"
+      )[0].innerHTML;
+      var indexFilter = tasks.indexOf(filterLabel);
+      var deleteElement = containerTaskActive[i].remove();
+    }
+  }
+}
+
+function filterCompletedTasks(e) {
+  var containerSelectButtonFilter = e.target.parentNode;
+  containerSelectButtonFilter.className = "filter_show bordered";
+  console.log("from completed" + containerSelectButtonFilter);
+
+  showTaskCompleted();
+  containerSelectButtonFilter.className = "filter_show";
+}
+
+function showTaskActive() {
+
+  showTaskAll();
+  console.log("afer rerendering");
+  console.log(tasks);
+  var containerTaskActive = document.querySelectorAll(".task_active");
+
+  for (var i = 0; i < containerTaskActive.length; i++) {
+    var childTask = containerTaskActive[i].firstElementChild;
+
+    if (childTask.matches(":checked")) {
+      var filterLabel = containerTaskActive[i].getElementsByClassName(
+        "task_label_checkbox"
+      )[0].innerHTML;
+      var indexFilter = tasks.indexOf(filterLabel);
+      var deleteElement = containerTaskActive[i].remove();
+    }
+  }
+}
+
+function filterActiveTasks(e) {
+  var containerSelectButtonFilter = e.target.parentNode;
+  containerSelectButtonFilter.className = "filter_show bordered";
+
+  showTaskActive();
+  containerSelectButtonFilter.className = "filter_show";
+}
+
+function showTaskAll() {
+
+  refreshAllTasks();
+
+  console.log("afer refreshing");
+  console.log(tasks);
+  var containerAllTasks = document.body.getElementsByClassName("tasks_all")[0];
```

Comment:
next step - try to implement "single responsibility" - each function should make just one task